File: 1468-check-if-n-and-its-double-exist/solution.go
func checkIfExist(arr []int) bool {
    memory := map[int]int{}

    for i, v := range arr {
        memory[v] = i
    }

    for i := 0; i < len(arr); i++ {
        if j, ok := memory[arr[i]*2]; ok && j != i {
            return true
        }
    }

    return false
}

File: 1468-check-if-n-and-its-double-exist/solution.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        n = len(arr)
        memo = set()
        # 2 * arr[i] => arr[j]
        # arr[j] / 2 = arr[i]
        for i in range(n):
            if 2 * arr[i] in memo:
                return True
            if arr[i] % 2 == 0 and arr[i] / 2 in memo:
                return True
            memo.add(arr[i])
        return False
